x-pack/plugins/cases/public/components/case_view/index.tsx

```diff
@@ -518,27 +519,63 @@ export const CaseView = React.memo(
-            <EuiLoadingSpinner data-test-subj="case-view-loading" size="xl" />
-          </EuiFlexItem>
-        </MyEuiFlexGroup>
-      );
-    }
-    if (onCaseDataSuccess && data) {
-      onCaseDataSuccess(data);
-    }
+    const { data, resolveOutcome, resolveAliasId, isLoading, isError, fetchCase, updateCase } =
+      useGetCase(caseId, subCaseId);
+    const { spaces: spacesApi, http } = useKibana().services;
 
-    return (
+    useEffect(() => {
+      if (onCaseDataSuccess && data) {
+        onCaseDataSuccess(data);
+      }
+    }, [data, onCaseDataSuccess]);
+
+    useEffect(() => {
+      if (spacesApi && resolveOutcome === 'aliasMatch' && resolveAliasId != null) {
+        // CAUTION: the path /cases/:detailName is working in both Observability (/app/observability/cases/:detailName) and
+        // Security Solutions (/app/security/cases/:detailName) plugins. This will need to be changed if this component is loaded
+        // under any another path, passing a path builder function by props from every parent plugin.
+        const newPath = http.basePath.prepend(
+          `cases/${resolveAliasId}${window.location.search}${window.location.hash}`
+        );
+        spacesApi.ui.redirectLegacyUrl(newPath, i18n.CASE);
+      }
+    }, [resolveOutcome, resolveAliasId, spacesApi, http]);
+
+    const getLegacyUrlConflictCallout = useCallback(() => {
+      // This function returns a callout component *if* we have encountered a "legacy URL conflict" scenario
+      if (data && spacesApi && resolveOutcome === 'conflict' && resolveAliasId != null) {
+        // We have resolved to one object, but another object has a legacy URL alias associated with this ID/page. We should display a
+        // callout with a warning for the user, and provide a way for them to navigate to the other object.
+        const otherObjectId = resolveAliasId; // This is always defined if outcome === 'conflict'
+        // CAUTION: the path /cases/:detailName is working in both Observability (/app/observability/cases/:detailName) and
+        // Security Solutions (/app/security/cases/:detailName) plugins. This will need to be changed if this component is loaded
+        // under any another path, passing a path builder function by props from every parent plugin.
+        const otherObjectPath = http.basePath.prepend(
+          `cases/${otherObjectId}${window.location.search}${window.location.hash}`
+        );
+        return spacesApi.ui.components.getLegacyUrlConflict({
+          objectNoun: i18n.CASE,
+          currentObjectId: data.id,
+          otherObjectId,
+          otherObjectPath,
+        });
+      }
+      return null;
+    }, [data, resolveAliasId, resolveOutcome, spacesApi, http.basePath]);
+
+    return isError ? (
+      <DoesNotExist allCasesNavigation={allCasesNavigation} caseId={caseId} />
+    ) : isLoading ? (
+      <MyEuiFlexGroup gutterSize="none" justifyContent="center" alignItems="center">
```

Comment:
Nit, nested ternaries can be a pain to reason about in the view, especially when components are added, or the ones here start to grow. That's why I think it was set up as if statements before. Might be worth changing it back or breaking each of these out into it's own components if you want to leave the ternaries so something like this:

```
return isError
	? <DoesNotExist ...props />
	: isLoading
	 ? <CasesLoadingComponent ...props />
	 : data && <CasesStuff ...props>
```

That last data check might be a little useless because if we don't get the data obj back it would just be an empty page, don't know if that ever happens though...

Reply:
I changed to the ternaries because before there was _return_s inside the if statements at the top, and I needed to use some hooks. To avoid using the ternaries we'd need to put all these if/return statements inside another `useMemo` and return it, which I don't think makes much sense since this component is already using a React.memo.
I think encapsulating the _isLoading_ component into a `<CaseViewLoading />` will improve readability. 

And regarding the `data` check, it is just a guard. It should never happen, but I feel better doing the condition than passing it with `caseData={data!}`, so I decided not to change that condition.